### PR TITLE
Implement IPC based on QIODevice and QSharedMemory

### PR DESCRIPTION
**Pull request by apopiak**

### CODE REVIEW

**Comment by apopiak** [comment]
@mrzzzrm check it out?

File: source/common/InterruptibleThread.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if (m_thread.joinable())
+    {
+        stop();
+        m_thread.join();
+    }
+}
+
+common::InterruptibleThread & InterruptibleThread::InterruptibleThread::operator=(common::InterruptibleThread && rhs)
+{
+    m_thread = move(rhs.m_thread);
```
Comment (mrzzzrm) [inline source/common/InterruptibleThread.cpp]: Is this std::move()? I would prefer typing out the namespace-specifier for std.

**Comment by mrzzzrm** [comment]
Looks fine to me.

**Comment by mrzzzrm** [comment]
:shipit:

**Comment by apopiak** [comment]
I don't know how it ended up with that message :-D

**Pull request merged**